plugins/reach/lib/model/EntryVendorTask.php

```diff
@@ -435,4 +435,22 @@ public function postSave(PropelPDO $con = null)
 		$this->indexToSearchIndex();
 	}
 
+	public function save(PropelPDO $con = null)
+	{
+		if (in_array(EntryVendorTaskPeer::STATUS, $this->modifiedColumns) &&
+			(in_array($this->status, array(EntryVendorTaskStatus::PROCESSING, EntryVendorTaskStatus::ABORTED))))
+		{
+			if (in_array($this->oldColumnsValues[EntryVendorTaskPeer::STATUS],
+				array(EntryVendorTaskStatus::PENDING, EntryVendorTaskStatus::PENDING_MODERATION, EntryVendorTaskStatus::PENDING_ENTRY_READY)))
+			{
+				parent::save($con);
+			}
+			else
+			{
+				throw new KalturaAPIException(KalturaReachErrors::ENTRY_VENDOR_TASK_ITEM_COULD_NOT_BE_UPDATED, $this->id, $this->status);
```

Comment:
can't raise API exception inside code.
Need to throw core exception here and in the api translate to KalturaAPIException